scene/2d/tile_map.cpp

```diff
@@ -805,33 +805,34 @@ void TileMap::_set_tile_data(const Vector<int> &p_data) {
 		if (format == FORMAT_3) {
 			uint16_t source_id = decode_uint16(&local[4]);
 			uint16_t atlas_coords_x = decode_uint16(&local[6]);
-			uint16_t atlas_coords_y = decode_uint32(&local[8]);
+			uint16_t atlas_coords_y = decode_uint16(&local[8]);
 			uint16_t alternative_tile = decode_uint16(&local[10]);
 			set_cell(Vector2i(x, y), source_id, Vector2i(atlas_coords_x, atlas_coords_y), alternative_tile);
 		} else {
 #ifndef DISABLE_DEPRECATED
 			uint32_t v = decode_uint32(&local[4]);
-			v &= (1 << 29) - 1;
-
-			// We generate an alternative tile number out of the the flags
-			// An option should create the alternative in the tileset for compatibility
```

Comment:
It doesn't seem immediately obvious what this bit shifting is about here, so I guess updating the comment would be good instead of removing it.